### ISSUE TITLE
[Question] ZFS snapshot is deleted along with the PVC

### ISSUE THREAD

**Issue by Subetov** [openebs/zfs-localpv#432]
**What steps did you take and what happened:**
I created a PCV, populated it with data, created a snapshot, then deleted the PCV, and it turned out that the snapshot also disappears from the host system and it can no longer be restored.
Although the resource "zfssnap" and "volumesnapshot" resources remain in kubernetes.

Is this expected behaviour?

**What did you expect to happen:**
I assumed that the snapshot will remain and I will be able to restore the data (create a new PVC from the snapshot)

**Environment:**
- ZFS-LocalPV version: 2.1.0
- Kubernetes version: v1.24.8

**Comment by sinhaashish** [comment]
hi @Subetov 
I tried to replicate this isuue, but i was not able to do . Please reply with the steps you followed.
Steps : followed 
1. Stopped the fio
2. Deleted the pvc
3. Checked the snapshot , it exist
![Screenshot 2023-09-13 at 10 50 55 AM](https://github.com/openebs/zfs-localpv/assets/13465383/4c1143e1-fb85-4757-a369-ec0e0732703c)

**Comment by Subetov** [comment]
Hello @sinhaashish 
Oh it was a long time ago, but as I remember snapshot was deleted from the host machine (not zfssnap k8s custom resource )
Checked with `zfs list -t snapshot` on the host machine.

**Comment by sinhaashish** [comment]
Hi @Subetov  ,
Yes its the expected behaviour. 
Upon deletion of PVC , the PV as well as snapshot gets deleted.
There is a PR which prevent volume deletion if snapshot exists . #350 . Its under review as of now.

**Comment by Subetov** [comment]
Thank you, @sinhaashish  for the reply.
I saw that ticket, it's been waiting its turn for two years now. =)

**Comment by sinhaashish** [comment]
@Subetov , yes there were some other concerns and high priority jobs for us, but now we are actively looking on this repo. 
Dont worry , your concern will be taken care.

**Comment by Subetov** [comment]
@sinhaashish, thank you a lot.

**Comment by avishnu** [comment]
Re-opening this for further investigation and fixing if needed.

**Comment by avishnu** [comment]
This seems important, prioritizing for v4.2

**Comment by sinhaashish** [comment]
Actually upon PVC deletion the underneath volume and snap shot were getting deleted. This is a bug .

There is a PR https://github.com/openebs/zfs-localpv/pull/613 to resolve this
